tools/fuzzer/ast.h

```diff
@@ -268,6 +269,29 @@ class TernaryExpr {
   std::unique_ptr<Expr> rhs_;
 };
 
+class BooleanConstant {
+ public:
+  enum class Value : bool { False, True };
+
+  static constexpr int PRECEDENCE = 0;
+
+  static constexpr Value True = Value::True;
+  static constexpr Value False = Value::False;
+
+  BooleanConstant(Value value) : value_(value) {}
+
+  friend std::ostream& operator<<(std::ostream& os,
+                                  const BooleanConstant& expr);
+
+  Value value() const { return value_; }
+  int precedence() const { return PRECEDENCE; }
+
+ private:
+  Value value_;
+};
+
+std::ostream& operator<<(std::ostream& os, BooleanConstant::Value value);
+
```

Comment:
Do you really need this indirection? Can you just print `BooleanConstant` directly (same as you're doing to all other nodes)?